src/libponyrt/sched/systematic_testing.h

```diff
@@ -0,0 +1,61 @@
+#ifndef sched_systematic_testing_h
+#define sched_systematic_testing_h
+
+#include <platform.h>
+#include "scheduler.h"
+
+#if defined(USE_SYSTEMATIC_TESTING)
+#if !defined(PLATFORM_IS_WINDOWS) && !defined(USE_SCHEDULER_SCALING_PTHREADS)
+pony_static_assert(false, "Systematic testing requires pthreads (USE_SCHEDULER_SCALING_PTHREADS) to be enabled!");
```

Comment:
> Why?

Because it was simpler to not worry about `signals` and rely only on `pthreads` to start with. In theory, it should be possible to extend this to using `signals` instead of `pthreads` when `USE_SCHEDULER_SCALING_PTHREADS` is not defined.

> Shouldn't it be possible to write this in such a way that it works both with and without scaling enabled?

i don't understand this question. The current implementation already works whether or not `scaling` is disabled or not via the (`--ponynoscale` cli option). Can you clarify what you meant by this question?

Reply:
At the time Joe asked the question, he had forgotten that you can turn on scaling and then one of two options for how the scaling is done. When we reviewed during sync today I reminded him of that.

Reply:
Got it. Thanks for the info.